java/arcs/core/analytics/Analytics.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.analytics
+
+import arcs.core.crdt.CrdtModel
+import arcs.core.crdt.CrdtSet
+import arcs.core.crdt.CrdtSingleton
+import arcs.core.storage.keys.Protocols
+import arcs.core.storage.referencemode.REFERENCE_MODE_PROTOCOL
+import arcs.core.util.TaggedLog
+
+/** Entry for logging analytics. */
+interface Analytics {
+
+    /** Types of Storage to log. */
+    enum class StorageType {
+        VOLATILE,
+        RAM_DISK,
+        MEMORY_DATABASE,
+        DATABASE,
+        REFERENCE_MODE,
+        OTHER
+    }
+
+    /** Types of Handles to log. */
+    enum class HandleType {
+        SINGLETON,
+        COLLECTION,
+        OTHER
+    }
+
+    /** Types of Storage events to log. */
+    enum class Event {
+        SYNC_REQUEST_TO_MODEL_UPDATE
+    }
+
+
+    /** Log storage latency based on [StorageType], [HandleType] and [Event]. */
+    fun logStorageLatency(
+        latencyMillis: Long,
+        storageType: StorageType,
+        handleType: HandleType,
+        event: Event
+    )
```

Comment:
Internal style prefers the functions to be listed before any inner classes or enums.

Reply:
Done.